great_expectations/expectations/expectation.py

```diff
@@ -2395,22 +2395,33 @@ def _format_map_output(
     if result_format["result_format"] == "BASIC":
         return return_obj
 
+    if len(unexpected_list) and isinstance(unexpected_list[0], dict):
+        # in the case of multicolumn map expectations `unexpected_list` contains dicts,
+        # which will throw an exception when we hash it to count unique members.
+        # As a workaround, we flatten the values out to tuples.
+        keys = unexpected_list[0].keys()
+        immutable_unexpected_list = [
+            tuple([el[key] for key in keys]) for el in unexpected_list
+        ]
```

Comment:
Just confirming - is the situation always that if `len(unexpected_list > 1)` the keys for all of the items of `unexpected_list` will be the same?

I'm wondering if instead it might be better to use`[tuple([v for v in i.values()]) for i in unexpected_list]` though I might not be reading this right.

Reply:
This is a good question and worth exploring. I believe end result would be the same, but your suggestions is cleaner.

Reply:
Code has been refactored and pushed. Really appreciate the suggestion.